dipy/workflows/multi_io.py

```diff
@@ -171,10 +172,17 @@ def io_iterator_(frame, fnc, output_strategy='append', mix_names=False):
     args.remove('self')
     del values['self']
 
-    specs = inspect.getargspec(fnc)
-    spargs = specs.args
-    spargs.remove('self')
-    defaults = specs.defaults
+    if sys.version_info[0] >= 3:
+        sig_object = inspect.signature(fnc)
```

Comment:
why you do not use `get_args_default`  in [`dipy/workflows/base.py`](https://github.com/nipy/dipy/blob/9242c796fa1b473c08911d3e59d85f9cc226bb95/dipy/workflows/base.py#L8)

I think it will avoid to duplicate the code